Cargo.toml

```diff
@@ -109,6 +115,13 @@ tcp = [
     "tokio/time",
 ]
 
+# Trace functionality
+traceable = [
```

Comment:
Is there appetite for "level" features, e.g. 'trace`, `debug`, `warn`. Such that one could conceivable build Hyper with zero tracing (runtime overhead), full tracing or anything in between.  The costs would be one off, at compile time, but they don't seem to be too large.

Reply:
We currently have features such as `client` and `server`.  

Those feature flags could be leveraged here too.

Furthermore, if we have features `incoming` and `outgoing` it would be possible to get quite targeted tracing data - avoiding being swamped by trace data.  This would in turn allow us to be more liberal in what is traced.

Example:

```bash
cargo run --features="traceable server incoming" --example echo_traceable
```